### PR TITLE
Add more examples of using URLSearchParams

### PR DESCRIPTION
**Pull request by codegirl-007**
This commit adds: 
- an example of using an object literal as a parameter.
- an explanation when you have duplicate parameters in a query string

> What was wrong/why is this fix needed? (quick summary only)
The documentation was missing some use cases of URLSearchParams.

> Issue number (if there is an associated issue)

> Anything else that could help us review it

### CODE REVIEW

**Review by wbamberg — CHANGES_REQUESTED** [review]
These examples look very helpful!

Although we're not good at being consistent with formatting in our examples, it's worth at least having consistent formatting within each example, so please could you:

* make sure all lines end with a semicolon (where appropriate of course)

* either update your new examples to use double quotes for strings, or (better but more work) update all the examples in this page to use single quotes?

Even better would be to use `let` rather than `var` for these examples (or `const` if you want to use different variables for the different examples), and to have a space between `//` and the comment itself. But those are optional :).

**Comment by stephanie-gredell** [comment]
@wbamberg Addressed your comments. Thanks for the suggestions. Let me know if there's anything else I could make better. :)

File: files/en-us/web/api/urlsearchparams/index.html [hunk 1]
```diff
@@ -60,57 +60,73 @@ <h2 id="Methods">Methods</h2>
...
-searchParams3.has("query") // true
+const url = new URL('http://example.com/search?query=%40');
+const searchParams3 = new URLSearchParams(url.search);
+searchParams3.has('query') // true
 
-let base64 = btoa(String.fromCharCode(19,224,23,64,31,128)); // base64 is "E+AXQB+A"
-let searchParams = new URLSearchParams("q=foo&bin=" + base64); // q=foo&bin=E+AXQB+A
-let getBin = searchParams.get("bin"); // "E AXQB A" + char is replaced by spaces
+const base64 = btoa(String.fromCharCode(19,224,23,64,31,128)); // base64 is "E+AXQB+A"
+const searchParams = new URLSearchParams('q=foo&bin='" + base64); // q=foo&bin=E+AXQB+A
```
Comment (wbamberg) [inline files/en-us/web/api/urlsearchparams/index.html:121]:
```suggestion
const searchParams = new URLSearchParams('q=foo&bin=' + base64); // q=foo&bin=E+AXQB+A
```

File: files/en-us/web/api/urlsearchparams/index.html [hunk 2]
```diff
@@ -60,57 +60,73 @@ <h2 id="Methods">Methods</h2>
...
-searchParams.get("foo") === null; // true
-searchParams.append("topic", "webdev");
+searchParams.has('topic') === true; // true
+searchParams.get('topic') === "api"; // true
+searchParams.getAll('topic'); // ["api"]
+searchParams.get('foo') === null; // true
+searchParams.append('topic', 'webdev');
 searchParams.toString(); // "q=URLUtils.searchParams&amp;topic=api&amp;topic=webdev"
-searchParams.set("topic", "More webdev");
+searchParams.set('topic', "More webdev");
```
Comment (wbamberg) [inline files/en-us/web/api/urlsearchparams/index.html:77]:
```suggestion
searchParams.set('topic', 'More webdev');
```

File: files/en-us/web/api/urlsearchparams/index.html [hunk 3]
```diff
@@ -60,57 +60,73 @@ <h2 id="Methods">Methods</h2>
 
 <h2 id="Examples">Examples</h2>
 
-<pre class="brush: js">var paramsString = "q=URLUtils.searchParams&amp;topic=api";
-var searchParams = new URLSearchParams(paramsString);
+<pre class="brush: js">var paramsString = 'q=URLUtils.searchParams&amp;topic=api';
```
Comment (wbamberg) [inline files/en-us/web/api/urlsearchparams/index.html:63]:
```suggestion
<pre class="brush: js">const paramsString = 'q=URLUtils.searchParams&amp;topic=api';
```

File: files/en-us/web/api/urlsearchparams/index.html [hunk 4]
```diff
@@ -60,57 +60,73 @@ <h2 id="Methods">Methods</h2>
...
+let paramsObj = {foo: 'bar', baz: 'bar'};
+let searchParams = new URLSearchParams(paramsObj);
+
+searchParams.toString(); // "foo=bar&baz=bar"
+searchParams.has('foo'); // true
+searchParams.get('foo'); // bar
+
+// Having duplicate parameters only returns the first value
+let paramStr = 'foo=bar&foo=baz';
+let searchParams = new URLSearchParams(paramsObj);
```
Comment (wbamberg) [inline files/en-us/web/api/urlsearchparams/index.html:92]:
```suggestion
let searchParams = new URLSearchParams(paramStr);
```

?

**Review by wbamberg — CHANGES_REQUESTED** [review]
Thanks for these updates! I fixed a couple of leftovers and made a couple of other suggestions.

Also though I think it's worth breaking this first example up into separate code blocks, because as it is, if someone copies that whole code block into the console they'll get an error about redeclaring `searchParams`.

So I think it would be worth having three separate code blocks:

* one from the start to `searchParams.toString(); // "q=URLUtils.searchParams"`
* one from `// Search parameters can also be an object` to `searchParams.get('foo'); // bar`
* one from `// Having duplicate parameters only returns the first value` to the end

I think this would also make it more obvious what each section is showing. What do you think?

**Comment by sideshowbarker** [comment]
@wbamberg I think this one might be waiting on your re-review

**Comment by wbamberg** [comment]
Sorry, I completely forgot about this. It looks like I was waiting for https://github.com/mdn/content/pull/7218#pullrequestreview-715375122 , but I just pushed a commit that addresses it, so let's merge :).

Thanks for the PR @stephanie-gredell , these are nice examples to have!

**Review by wbamberg — APPROVED** [review]

**Comment by stephanie-gredell** [comment]
> Sorry, I completely forgot about this. It looks like I was waiting for https://github.com/mdn/content/pull/7218#pullrequestreview-715375122 , but I just pushed a commit that addresses it, so let's merge :).
> 
> 
> 
> Thanks for the PR @stephanie-gredell , these are nice examples to have!

Thank you for the help in the contribution! Life got busy and I also forgot to follow. Cheers!

**Pull request merged**